src/Xrefcheck/Verify.hs

```diff
@@ -699,15 +722,18 @@ checkExternalResource Config{..} link
 
       let maxTime = Time @Second $ unTime ncExternalRefCheckTimeout * timeoutFrac
 
-      mres <- liftIO (timeout maxTime $ void reqLink) `catch`
-        (either throwError (\() -> return (Just ())) . interpretErrors)
-      maybe (throwError $ ExternalResourceSomeError "Response timeout") pure mres
-
-    isTemporaryRedirectCode :: Int -> Bool
-    isTemporaryRedirectCode = flip elem [302, 303, 307]
+      reqRes <- catch (liftIO (timeout maxTime $ reqLink $> RRDone)) $ \httpErr ->
+        case interpretErrors httpErr of
+          Left err -> throwError err
+          Right res -> pure $ Just res
 
-    isPermanentRedirectCode :: Int -> Bool
-    isPermanentRedirectCode = flip elem [301, 308]
+      case reqRes of
+        Nothing -> throwError $ ExternalResourceSomeError "Response timeout"
+        Just RRDone -> pure ()
+        Just (RRFollow nextLink) -> do
+          VerifyResult errs <- liftIO $
+            checkExternalResource (followed `pushRequest` link) config nextLink
+          mapM_ throwError errs
```

Comment:
:trollface: What the hell is going on here? Don't we just throw the first error and ignore all others?

Reply:
That is right, but `checkExternalResource` returns `VerifyResult VerifyError` (i.e. a list of `VerifyError`) and this is `ExceptT VerifyError IO ()`. I also saw that `checkExternalResource` does not produce ever more than one error.

In the future, if `checkExternalResource` produces more than one error, some of them may be missed when following redirects. Maybe we can just change `checkExternalResource` to return `Maybe VerifyError`, or this `ExceptT VerifyError IO ()` to be `ExceptT (VerifyResult VerifyError) IO ()` instead.

What do you think? If I'm not wrong, you were also refactoring these error types and monads in some of your PRs.

Reply:
Also, I think that this would be surely changed in #260, an issue that has been created due to a related discussion in this same PR.

Reply:
I think for now it's OK to just throw the first error, in [234](https://github.com/serokell/xrefcheck/pull/234) all the types are changed such that only one error was possible, and that pull is ready to merge.